### PR TITLE
Validator ajv6

### PR DESCRIPTION
**Pull request by heath-freenome**
### Reasons for making this change

Partially fixes https://github.com/rjsf-team/react-jsonschema-form/issues/2693 by migrating the validator into its own package.

Implements the `@rjsf/validator-ajv6` package as a port of the `core/utils/validator.js` file, converting things into a class, a type and some helper functions.

All of the migrated code in this PR have been converted to typescript with 100% unit testing
Added full documentation for the converted functions

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/validator-ajv6/src/createAjvInstance.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  customFormats?: CustomValidatorOptionsType['customFormats'],
+) {
+  const ajv = new Ajv(AJV_CONFIG);
+
+  // add custom formats
+  ajv.addFormat('data-url', DATA_URL_FORMAT_REGEX);
+  ajv.addFormat('color', COLOR_FORMAT_REGEX);
+
+  // add more schemas to validate against
+  if (additionalMetaSchemas && Array.isArray(additionalMetaSchemas)) {
```
Comment (heath-freenome) [inline packages/validator-ajv6/src/createAjvInstance.ts:34]: `additionalMetaSchemas &&` is redundant

File: packages/validator-ajv6/src/createAjvInstance.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ajv.addMetaSchema(additionalMetaSchemas);
+  }
+
+  // add more custom formats to validate against
+  if (customFormats && isObject(customFormats)) {
```
Comment (heath-freenome) [inline packages/validator-ajv6/src/createAjvInstance.ts:39]: `customFormats &&` is redundant

**Comment by heath-freenome** [comment]
@nickgros ready for review

**Review by nickgros — APPROVED** [review]

File: packages/validator-ajv6/test/utilsTests/getTestValidator.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    ): ValidationData<T> {
+      return validator.validateFormData(formData, schema, customValidate, transformErrors);
+    },
+    toErrorList (errorSchema?: ErrorSchema<T>, fieldName?: string): RJSFValidationError[] {
+      return validator.toErrorList(errorSchema, fieldName);
+    },
+    isValid (schema: RJSFSchema, formData: T, rootSchema: RJSFSchema): boolean {
+      return validator.isValid(schema, formData, rootSchema);
+    },
+    setReturnValues() {
```
Comment (heath-freenome) [inline packages/validator-ajv6/test/utilsTests/getTestValidator.ts:27]: Add comment about this intentionally being a noop

File: packages/validator-ajv6/src/validator.ts [hunk 4]
```diff
@@ -210,7 +210,7 @@ export default class AJV6Validator<T = any> implements ValidatorType<T> {
   ): ValidationData<T> {
     // Include form data with undefined values, which is required for validation.
     const rootSchema = schema;
-    const newFormData = getDefaultFormState(this, schema, formData, rootSchema, true);
+    const newFormData = getDefaultFormState(this, schema, formData, rootSchema, true) as T;
```
Comment (heath-freenome) [inline packages/validator-ajv6/src/validator.ts:213]: Add generic to function and remove the cast

**Review by nickgros — APPROVED** [review]

**Pull request merged**